### PR TITLE
Apple silicon support

### PR DESCRIPTION
**Pull request by Baltoli**
This PR implements Apple Silicon support for the K framework. The changes required to do so are as follows:

* Update Jansi dependency to `2.4.0`
  * Jansi uses native code bindings to call system functions like `isatty`; these bindings are precompiled and distributed with the library.
  * `2.4.0` is the first version that distributes a working native library for macOS on ARM.
* Update Jline dependency to `3.21.0`
  * Jline bundles Jansi, and so we need to update to a newer version of Jline to avoid transitively dragging in an older version of Jansi (this is the cause of the [issue noted here](https://github.com/kframework/k/issues/2276#issuecomment-947126681) by @Acassimiro).
  * Making this change entails some minor API changes where we use Jline and Jansi; none of these are complex.
* Adds a line to the `haskell-backend` POM to pass in the additional options to stack that are currently required to build the backend (see [the documentation](https://github.com/kframework/kore#apple-silicon) and [associated PR](https://github.com/kframework/kore/pull/2893) for more details).
* For ease of working on this PR on my M1 machine, it pulls in the build system changes in https://github.com/kframework/llvm-backend/pull/443.
  * ~~The backend PR should be merged and the submodule updated _before_ this PR is, as the build system changes to the backend shouldn't affect anything not on Apple Silicon in the frontend.~~
  * ~~Then, this PR can be merged once the submodule is updated.~~
* ~~This PR shouldn't be merged until the MPFR dependency has been separately pushed to Maven and updated.~~

### CODE REVIEW

**Comment by Baltoli** [comment]
Blocked until #2290 is merged.

**Comment by Baltoli** [comment]
@dwightguth This PR doesn't produce a perfectly-working toolchain on ARM, by the way - there are a few minor issues I'm going to write up as separate fixes in the interest of:
<ol type="a">
  <li>Keeping this PR to a manageable size and single focus (i.e. fixing the _build_ issues for Apple Silicon)</li>
  <li>Merging to produce a mostly-working toolchain that people are able to use before we have CI requirements</li>
</ol>

**Review by dwightguth — APPROVED** [review]

**Pull request merged**
